### PR TITLE
feat(mce-consumer): removing dependency on gms for mce consumer

### PR DESCRIPTION
**Pull request by david-leifker**
Adding envs preparing for v0.9.6 mce consumer updates

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Review by pedro93 — APPROVED** [review]
LGTM, please merge master and fix conflicts

File: charts/datahub/subcharts/datahub-mce-consumer/templates/deployment.yaml [hunk 1]
```diff
@@ -101,10 +101,64 @@ spec:
...
+            {{- with .Values.global.elasticsearch.auth }}
+            - name: ELASTICSEARCH_USERNAME
+              value: {{ .username }}
+            - name: ELASTICSEARCH_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ .password.secretRef }}"
+                  key: "{{ .password.secretKey }}"
+            {{- end }}
+            {{- with .Values.global.elasticsearch.indexPrefix }}
```
Comment (szalai1) [inline charts/datahub/subcharts/datahub-mce-consumer/templates/deployment.yaml:143]: why having with here is better than just using this as value 2lines below?

File: charts/datahub/subcharts/datahub-mce-consumer/templates/deployment.yaml [hunk 2]
```diff
@@ -101,10 +101,64 @@ spec:
...
+            # TODO: Remove DATAHUB_GMS_HOST/PORT after v0.9.6
             - name: DATAHUB_GMS_HOST
               value: {{ printf "%s-%s" .Release.Name "datahub-gms" }}
             - name: DATAHUB_GMS_PORT
               value: "{{ .Values.global.datahub.gms.port }}"
+            - name: ENTITY_REGISTRY_CONFIG_PATH
+              value: /datahub/datahub-mce-consumer/resources/entity-registry.yml
+            - name: EBEAN_DATASOURCE_USERNAME
+              value: "{{ .Values.global.sql.datasource.username }}"
+            - name: EBEAN_DATASOURCE_PASSWORD
```
Comment (szalai1) [inline charts/datahub/subcharts/datahub-mce-consumer/templates/deployment.yaml:113]: we support password as value for other resources, we should do it here too for consistency

Comment (szalai1) [reply charts/datahub/subcharts/datahub-mce-consumer/templates/deployment.yaml:113]: like here: https://github.com/acryldata/datahub-helm/blob/master/charts/datahub/templates/datahub-upgrade/_upgrade.tpl#L20

**Review by szalai1 — APPROVED** [review]
👍

File: charts/datahub/subcharts/datahub-mce-consumer/templates/deployment.yaml (hunk 1 above)
Comment (david-leifker) [reply charts/datahub/subcharts/datahub-mce-consumer/templates/deployment.yaml:147]: This is consistent with gms is all.

**Pull request merged**